R/Prediction.R

```diff
@@ -195,6 +195,57 @@ makePrediction.CostSensTaskDesc = function(task.desc, row.names, id, truth, pred
 }
 
 #' @export
+makePrediction.ForecastRegrTaskDesc = function(task.desc, row.names, id, truth, predict.type,
+                                               predict.threshold = NULL, y, time, error = NA_character_, dump = NULL) {
+  data = namedList(c("id", "truth", "response", "se"))
+  if (any(class(y) == "matrix"))
+    size.y = nrow(y)
+  else
+    size.y = length(y)
+
+  # This will only happen when there is a task with no subset
+  #  aka, we predict future values and have to get their times
+  if (length(truth) > size.y) {
+    sec = lubridate::dseconds(lubridate::int_diff(as.POSIXct(row.names)))
+    # We take the median seconds between intervals as this won't get
+    # a wrong day until about 200 months in the future.
+    med_sec = mean(sec)
+    start = as.POSIXct(row.names[length(row.names)])
+    row.names = start + rep(med_sec,size.y) * 1:size.y
```

Comment:
try to avoid variable names that are also functions, use e.g. `rn`  instead of `row.names` (the latter is also a function)